[DM-27342] Fix some specification errors

rra opened the pull request:
Fix a few things that turned up during implementation:

- scopes may be NULL in SQL since using an empty string is weird
- Specify how to return a new token
- Add last_used to the token info
- Improve the index used for finding existing internal tokens

rra committed:
Fix some specification errors

Fix a few things that turned up during implementation:

- scopes may be NULL in SQL since using an empty string is weird
- Specify how to return a new token
- Add last_used to the token info
- Improve the index used for finding existing internal tokens

ktlim commented on index.rst:
```diff
@@ -528,6 +528,13 @@ For all routes listed below with a ``username`` path parameter, only administrat
+    .. code-block:: json
+
+       {
+         "token": "gt-qVGZIh65TAJlNprOaMDhwg.WlUA5zyAY16dDRvDYxnwhg"
```
Is it `gt` or `gsh`?

rra replied:
`gt-`, sorry. Fixed in the next version.

ktlim reviewed (commented):
A few other questions:

- Why not have `child` be the primary key in the `subtoken` table?  They should be unique, right?
- Since usernames are not validated, an admin could remove all valid users but leave an invalid one.  We don't need to prevent this foot-shooting, but maybe the UI should warn against it when deleting the current user from the `admin` table.
- Should we anonymize/mask IP addresses at some point?

rra commented:
- Good call, yes, `child` can be the primary key. Made that fix.
- The admin handling is a temporary stopgap until we have the group component up and running. I've made that more explicit and added a note to the admins API.
- Good question on IP redaction. It makes them a lot less useful for analyzing problems, but it would also reduce the risk of the database being compromised somewhat, and possibly would help with GDPR for deployments that need that. I'm not sure what to do about that and probably need to do more reading on GDPR, but I'll add a note.

rra committed:
Fixes based on review

- Add a note about IP address retention
- Use gt- uniformly, replacing remnant gsh- mentions
- Note that the admins API is a temporary stopgap
- Make child the primary key of the subtokens table
